tests/python/test_gui.py

```diff
@@ -0,0 +1,30 @@
+from taichi import make_temp_file
+import taichi as ti
+import numpy as np
+import pytest
+
+
+@ti.host_arch_only
+@pytest.mark.parametrize('dtype', [ti.u8, ti.f32])
+def test_save_image_without_window(dtype):
+    n = 255
+    pixels = ti.field(dtype=dtype, shape=(n, n, 3))
+
+    @ti.kernel
+    def paint(c: dtype):
+        for i, j, k in pixels:
+            pixels[i, j, k] = c
+
+    gui = ti.GUI("Test", show_gui=False, res=(n, n))
```

Comment:
```suggestion
    gui = ti.GUI("Test", (n, n), show_gui=False)
```

Reply:
I think it would be a good idea to keep the named argument for `res` in case of constructor order changing?

Reply:
For example, `examples/stable_fluid.py` is using `ti.GUI('Stable Fluid', (res, res))`, so it's already broken by [this change](https://github.com/taichi-dev/taichi/pull/1817#discussion_r481673588). I believe many end-users are using `res` as positional argument too.
No matter if positional argument is a good design or not, we always want to enforce backward compatibility as much as possible.

Reply:
[Here](https://github.com/taichi-dev/taichi/pull/1817/files#diff-d10040c6d1ae795e8ecdaf6616941ab5R34-R41), I've moved `show_gui` to the last argument in the constructor now so that the original order is preserved:

```
def __init__(self,
                 name='Taichi',
                 res=512,
                 background_color=0x0,
                 show_gui=True):
```